### DISCUSSION TITLE
conanfile usage

### DISCUSSION THREAD

**Discussion by forgottosave** [ArthurSonzogni/FTXUI#823] [Q&A]
Hello :)
I have a question regarding the conanfile integration. I use the ftxui/5.0.0 requirement and implement it just like I do with GTest.

## Error
After calling `conan install .. --build=missing` in the build directory everything works and all `ftxui[...].cmake` get placed in `/build/Release/generators` (just like all `GTest[...].cmake` files).
Unfortunately `cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_VERBOSE_MAKEFILE=ON ..` results in the following error message:
![image](https://github.com/ArthurSonzogni/FTXUI/assets/88790311/b6dfd66c-b52b-4cc2-a613-da30b77794ba)

Do you have any idea, what I might have done wrong? Should there be a `Findftxui.cmake`, just like there is for GTest? Or is there any example-project using a conanfile with ftxui?

I wasn't sure if I should create an issue for this, since it can be, that I'm just too confused to use it properly ;)

## "Attachments"
important files & structures used (for clarification)
**project structure**
![project structure](https://github.com/ArthurSonzogni/FTXUI/assets/88790311/1f02d919-3d9a-499b-9aaa-2baf3a52697a)

**conanfile**
```conanfile.txt
[requires]
gtest/1.13.0
ftxui/5.0.0

[generators]
CMakeDeps
CMakeToolchain

[layout]
cmake_layout
```

**CMakeLists.txt (top)**
```CMakeLists.txt
cmake_minimum_required(VERSION 3.10)

project(CMakeExample)

set(CMAKE_CXX_STANDARD 17)

set(CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}" "${CMAKE_MODULE_PATH}")
set(CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}" "${CMAKE_PREFIX_PATH}")

message(${CMAKE_PREFIX_PATH})

set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib")
set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")

option(BUILD_TESTS "Build tests" ON)

enable_testing()

add_subdirectory(src)
add_subdirectory(app)

if (BUILD_TESTS)
    add_subdirectory(tests)
endif ()
```

**CMakeLists.txt (where ftxui is used)**
```app/CMakeLists.txt
find_package(ftxui REQUIRED)

add_executable(ostree-tui
    main.cpp
)

target_link_libraries(ostree-tui
    PRIVATE
    commandline
    manager
)
```

**Comment by ArthurSonzogni** [comment]
Thanks!

I never used Conan.
Here is a repo using conan+FTXUI:
https://github.com/theqbz/csv_helper/blob/main/src/CMakeLists.txt#L57-L59
Does is fixes your issue?

Alternatively, I see from the conan tutorial that you need to pass "-DCMAKE_TOOLCHAIN_FILE=conan_toolchain.cmake" for cmake to understand what conan setup.
